src/components/views/ViewMilestone.jsx

```diff
@@ -160,6 +162,34 @@ class ViewMilestone extends Component {
     );
   }
 
+  async shouldShowDelegateButton(isDelegate) {
+    const { currentUser } = this.props;
+    const { campaign } = this.state;
+    if (!campaign || !currentUser) {
+      return;
+    }
+    // the first page loaded maybe user or campaign be undefined, so we check if there
+    // aren't undefined proceed checking
+    this.setState({ checkShowDelegation: true });
+    const userAddress = currentUser && currentUser._address;
+    if (campaign.ownerAddress === userAddress) {
+      this.setState({ showDelegateButton: true });
+      return;
+    }
+
+    // should check if user has any dac
+    const userDacs = (await DACService.getDACsOwnedByUser(userAddress)).data;
```

Comment:
I imagined moving all this logic to user provider too.

Reply:
@aminlatifi  Ooops you are right, it can be moved too.

Reply:
@aminlatifi 
Now all logic moved to user provider.